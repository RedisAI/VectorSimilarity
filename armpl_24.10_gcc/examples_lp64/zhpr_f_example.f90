!   Double complex hermitian rank 1 update in packed format
!
!   Arm Performance Libraries version 24.10
!   SPDX-FileCopyrightText: Copyright 2015-2024 Arm Limited and/or its affiliates
!
    Program zhpr_f_example

      Use armpl_library
      Use, Intrinsic                   :: iso_fortran_env, Only: wp => real64

      Implicit None

      Character (Len=1), Parameter     :: uplo = 'U'

      Integer, Parameter               :: n = 4
      Integer, Parameter               :: incx = 1

      Real (Kind=wp), Parameter        :: alpha = 1.1_wp

      Complex (Kind=wp), Dimension (n*(n+1)/2) :: ap
      Complex (Kind=wp), Dimension (n) :: x

      Integer                          :: i, j, idx

      Write (*, 99999) 'ARMPL example: double complex hermitian &
        &rank 1 operation using ZHPR'
      Write (*, 99999) '-------------------------------------------------------&
        &------------'
      Write (*, *)

!
!     Initialize matrix A in packed format
      Do i = 1, n*(n+1)/2
        ap(i) = cmplx(i, i+1, kind=wp)
      End Do
!
!     Initialize vector x and y
      Do i = 1, n
        x(i) = cmplx(2*i-1, 2*i, kind=wp)
      End Do
!
      Write (*, 99998) 'alpha : ', alpha
      Write (*, *)
      Write (*, 99999) 'Matrix A:'
      Do i = 1, n
        Do j = 1, n
          If (j<i) Then
            idx = j + i*(i-1)/2
            Write (*, 99997, Advance='no') real(ap(idx)), -aimag(ap(idx))
          Else If (j>i) Then
            idx = i + j*(j-1)/2
            Write (*, 99997, Advance='no') real(ap(idx)), aimag(ap(idx))
          Else If (j==i) Then
            idx = i + j*(j-1)/2
            Write (*, 99997, Advance='no') real(ap(idx)), 0.0_wp
          End If
        End Do
        Write (*, *)
      End Do
!
      Write (*, *)
      Write (*, 99999) 'Vector x:'
      Do i = 1, n
        Write (*, 99997) x(i)
      End Do
!
      Call zhpr(uplo, n, alpha, x, incx, ap)
!
      Write (*, *)
      Write (*, 99999) 'Updated matrix A = alpha*x*x**H + A:'
      Do i = 1, n
        Do j = 1, n
          If (j<i) Then
            idx = j + i*(i-1)/2
            Write (*, 99997, Advance='no') real(ap(idx)), -aimag(ap(idx))
          Else If (j>=i) Then
            idx = i + j*(j-1)/2
            Write (*, 99997, Advance='no') real(ap(idx)), aimag(ap(idx))
          End If
        End Do
        Write (*, *)
      End Do
!
99999 Format (A)
99998 Format (A, F6.3)
99997 Format (1X, 1P, E10.3, Sp, E10.3, 'i')
!
    End Program
