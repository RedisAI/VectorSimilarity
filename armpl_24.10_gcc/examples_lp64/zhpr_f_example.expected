ARMPL example: double complex hermitian rank 1 operation using ZHPR
-------------------------------------------------------------------

alpha :  1.100

Matrix A:
  1.000E+00+0.000E+00i  2.000E+00+3.000E+00i  4.000E+00+5.000E+00i  7.000E+00+8.000E+00i
  2.000E+00-3.000E+00i  3.000E+00+0.000E+00i  5.000E+00+6.000E+00i  8.000E+00+9.000E+00i
  4.000E+00-5.000E+00i  5.000E+00-6.000E+00i  6.000E+00+0.000E+00i  9.000E+00+1.000E+01i
  7.000E+00-8.000E+00i  8.000E+00-9.000E+00i  9.000E+00-1.000E+01i  1.000E+01+0.000E+00i

Vector x:
  1.000E+00+2.000E+00i
  3.000E+00+4.000E+00i
  5.000E+00+6.000E+00i
  7.000E+00+8.000E+00i

Updated matrix A = alpha*x*x**H + A:
  6.500E+00+0.000E+00i  1.410E+01+5.200E+00i  2.270E+01+9.400E+00i  3.230E+01+1.460E+01i
  1.410E+01-5.200E+00i  3.050E+01+0.000E+00i  4.790E+01+8.200E+00i  6.630E+01+1.340E+01i
  2.270E+01-9.400E+00i  4.790E+01-8.200E+00i  7.310E+01+0.000E+00i  1.003E+02+1.220E+01i
  3.230E+01-1.460E+01i  6.630E+01-1.340E+01i  1.003E+02-1.220E+01i  1.343E+02+0.000E+00i
